### ISSUE TITLE
trimRight & trimLeft are not part of the ES standard

### ISSUE THREAD

**Issue by lukelex** [DavidAnson/markdownlint#99]
I've stumbled on a use case where polyfills are not an option. Ideally I'd suggest rewriting all ocurrencies into using inline functions instead of relying on the client to have it available.

I'm willing to implement this if you agree.

For reference:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/TrimRight
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/TrimLeft

**Comment by DavidAnson** [comment]
Interesting timing, I did this a few days ago: https://github.com/DavidAnson/markdownlint/commit/1184281c87faf2c79026c1ff6d4590b6f39af784

It will be part of the next release, probably early January. Is that okay for you?

**Comment by lukelex** [comment]
I'll be fine for now since I'm using a [forked](https://github.com/lukelex/markdownlint/commit/2889490e5165186bc287261348bf45d7edf054c3) version with a similar approach.

On another note, It'd be nice to know wether you have plans for the project so that noobs like myself can collaborate in ways that you'd be willing to accept.

**Comment by DavidAnson** [comment]
I generally try to work through the issue list, prioritizing bugs. The bulk of issues now are new rules, some of which I’m not sure I agree with, at least not defaulted to enabled which is convention.

What kind of planning do you have in mind?

**Comment by lukelex** [comment]
Maybe some general thoughts on upcoming features that are yet to be implemented for future versions. Maybe consider grouping them into a milestone for the next version or tag them with `up-for-grabs`.

**Comment by DavidAnson** [comment]
I don’t have any significant plans that aren’t already reflected in the issue list. Right now, the thing that seems most valuable is enabling custom rules to be added at run time. This will allow anyone to develop and refine any of the proposed rules without waiting on me. I had been deferring this work for a variety of reasons, but recently came up with an approach I think I like. If successful, contributors could pick up any of the rules in the issues list and implement them independently.

**Comment by DavidAnson** [comment]
Released in `0.7.0`.
